android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java

```diff
@@ -131,10 +135,13 @@ public void onReceived() throws InvalidNotificationException {
             }
         }
 
-        if ("clear".equals(type)) {
-            cancelNotification(data, notificationId);
-        } else {
+        switch(type) {
+        case PUSH_TYPE_MESSAGE:
             super.postNotification(notificationId);
+            break;
+        case PUSH_TYPE_CLEAR:
+            cancelNotification(data, notificationId);
```

Comment:
Dont you need to cancel it for update_badge too?

Reply:
The name `cancelNotification` is actually a bit misleading since it's function is to clear an existing notification.

As long as we don't call `postNotification` to create a visible notification, we don't need to do anything to have it ignore the `update_badge` notification. That's the main change here since previously this code would've called `postNotification` when it received an `update_badge`.